src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -34,30 +35,35 @@ const NewQuestionnaireScreen = () => {
   const { questionnairesApi } = useLambdasApi();
   const [loading, setLoading] = useState(false);
   const setError = useSetAtom(errorAtom);
+  const [tooltipMessage, setTooltipMessage] = useState<string>("");
+  const [isDisabled, setIsDisabled] = useState<boolean>(true);
   const [questionnaire, setQuestionnaire] = useState<Questionnaire>({
     title: "",
     description: "",
     questions: [],
     passScore: 0
   });
-  const checkBeforeSave = !questionnaire.title.trim() || !questionnaire.description.trim();
 
   /**
    * Function to handle input change in the questionnaire title and description
-   * 
+   *
    * @param event
    */
   const handleQuestionnaireInputChange = (event: ChangeEvent<HTMLInputElement>) => {
     const { name, value } = event.target;
-    setQuestionnaire((prevQuestionnaire) => ({
-      ...prevQuestionnaire,
-      [name]: value
-    }));
+    setQuestionnaire((prevQuestionnaire) => {
+      const updatedQuestionnaire = {
+        ...prevQuestionnaire,
+        [name]: value,
+      } as Questionnaire;
+      updateTooltipAndIsDisabled();
```

Comment:
Does this need to be called in the body of the set state? Can't it be called after the setQuestionnaire but still in the handleQuestionnaireInputChange fucntion?